gqless/src/QueryBuilder/SelectionTree/getAlias.ts

```diff
@@ -3,27 +3,44 @@ import { FieldSelection } from '../../Selection'
 import { createMemo } from '@gqless/utils'
 
 const memoized = createMemo()
+let id = 0
 
 export const getAlias = (tree: SelectionTree<FieldSelection>) => {
   if (!tree.parent) return
 
-  const fieldAliases = memoized(() => {
-    const aliases = new Map<FieldSelection, string>()
+  for (const siblingTree of tree.parent.children) {
```

Comment:
This is the sketchy part because it's doing two things: undoing the memo, which fixed a bug in our production build, but also naming everything. It likely should be redone, I'm going to give it a look again.